tests/linearGaussian_npse_test.py

```diff
@@ -1,300 +1,226 @@
-        )
-        target_samples = gt_posterior.sample((num_samples,))
-    elif prior_type == "uniform":
-        target_samples = samples_true_posterior_linear_gaussian_uniform_prior(
-            x_o,
-            likelihood_shift,
-            likelihood_cov,
-            prior,  # type: ignore
+            test_case.x_o,
+            test_case.likelihood_shift,
+            test_case.likelihood_cov,
+            test_case.prior_mean,
+            test_case.prior_cov,
         )
 
-    # Compute the c2st and assert it is near chance level of 0.5.
-    # Some degradation is expected, also because posterior get tighter which
-    # usually makes the c2st worse.
-    check_c2st(
-        samples,
-        target_samples,
-        alg=f"npse-{sde_type}-{prior_type}-{num_dim}-{iid_method}-{num_trial}iid-trials",
-        tol=0.05 * min(num_trial, 8),
-    )
+        map_ = posterior.map(show_progress_bars=True, num_iter=5)
+        assert ((map_ - gt_posterior.mean) ** 2).sum() < 0.5, "MAP is not close to GT."
 
 
 @pytest.mark.slow
-def test_npse_map():
-    num_dim = 2
-    x_o = zeros(num_dim)
-    num_simulations = 3000
-
-    # likelihood_mean will be likelihood_shift+theta
-    likelihood_shift = -1.0 * ones(num_dim)
-    likelihood_cov = 0.3 * eye(num_dim)
-
-    prior_mean = zeros(num_dim)
-    prior_cov = eye(num_dim)
-    prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
-    gt_posterior = true_posterior_linear_gaussian_mvn_prior(
-        x_o, likelihood_shift, likelihood_cov, prior_mean, prior_cov
+@pytest.mark.parametrize("npse_trained_model", test_cases_gaussian, indirect=True)
+def test_kld_gaussian(npse_trained_model):
+    # For the Gaussian prior, we compute the KLd between ground truth and
+    # posterior.
+    inference, score_estimator, test_case = npse_trained_model
+    posterior = inference.build_posterior(score_estimator)
+    posterior.set_default_x(test_case.x_o)
+    dkl = get_dkl_gaussian_prior(
+        posterior,
+        test_case.x_o[0],
+        test_case.likelihood_shift,
+        test_case.likelihood_cov,
+        test_case.prior_mean,
+        test_case.prior_cov,
+    )
+    max_dkl = 0.15
+    assert dkl < max_dkl, (
+        f"D-KL={dkl} is more than 2 stds above the average performance."
     )
-    inference = NPSE(prior, show_progress_bars=True)
-
-    theta = prior.sample((num_simulations,))
-    x = linear_gaussian(theta, likelihood_shift, likelihood_cov)
-
-    inference.append_simulations(theta, x).train()
-    posterior = inference.build_posterior().set_default_x(x_o)
 
-    map_ = posterior.map(show_progress_bars=True, num_iter=5)
 
-    assert ((map_ - gt_posterior.mean) ** 2).sum() < 0.5, "MAP is not close to GT."
+@pytest.mark.parametrize("test_case", test_cases_all)
+@pytest.mark.parametrize("sample_with", sample_with)
+@pytest.mark.parametrize("iid_method", iid_methods)
+def test_npse_iid_inference_snapshot(
```

Comment:
great! 

But this is the only snapshot know, right? and the idea is that is covers the entire npse workflow? 

because I was expecting to see more snapshots, e.g., one for the plain NPSE inference without iid-setting.